.github/workflows/publish_docker.yaml

```diff
@@ -62,37 +56,60 @@ jobs:
+      # skipping all of the tasks that don't match. This leaves a ton of "skipped" jobs
+      # but is the fastest way to get this working without overhauling the tag check logic.
+      matrix:
+        application: ["fides", "sample_app", "privacy_center"]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # This is required to properly tag images
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ env.DOCKER_USER }}
+          password: ${{ env.DOCKER_TOKEN }}
+
+      - name: Install Dev Requirements
+        run: pip install -r dev-requirements.txt
+
       # if neither prod, rc, beta or alpha git tag, then push images with the ":dev" tag
       - name: Push Fides Dev Tag
-        if: steps.check-prod-tag.outputs.match == 'false' && steps.check-rc-tag.outputs.match == 'false' && steps.check-beta-tag.outputs.match == 'false' && steps.check-alpha-tag.outputs.match == 'false'
-        run: nox -s "push(dev)"
+        if: needs.ParseTags.outputs.prod_tag == 'false' && needs.ParseTags.outputs.rc_tag == 'false' && needs.ParseTags.outputs.beta_tag == 'false' && needs.ParseTags.outputs.alpha_tag == 'false'
+        run: nox -s "push(${{ matrix.application }},dev)"
 
       # if a prod git tag, then we run the prod job to publish images tagged with the version number and a constant ":latest" tag
       - name: Push Fides Prod Tags
-        if: steps.check-prod-tag.outputs.match == 'true'
-        run: nox -s "push(prod)"
+        if: needs.ParseTags.outputs.prod_tag == 'true'
+        run: nox -s "push(${{ matrix.application }},prod)"
 
       # if an RC git tag, then we run the rc job to publish images with an ":rc" tag
       - name: Push Fides RC Tags
-        if: steps.check-rc-tag.outputs.match == 'true'
-        run: nox -s "push(rc)"
+        if: needs.ParseTags.outputs.rc_tag == 'true'
+        run: nox -s "push(${{ matrix.application }},rc)"
+
+      # if an alpha or beta git tag, then we run the prerelease job to publish images with an ":prerelease" tag
+      - name: Push Fides prerelease Tags
+        if: needs.ParseTags.outputs.alpha_tag == 'true' || needs.ParseTags.outputs.beta_tag == 'true'
+        run: nox -s "push(${{ matrix.application }},prerelease)"
+
+      # if not a prod git tag, then we run the git-tag job to publish images with a git tag
+      # if one exists on the current commit. the job is a no-op if the commit hasn't been tagged
+      - name: Push Fides Commit Tags
+        if: needs.ParseTags.outputs.prod_tag == 'false'
+        run: nox -s "push(${{ matrix.application }},git-tag)"
 
       # if an RC git tag, also notify Fidesinfra to trigger a redeploy of rc env, to pick up our newly published images
       - name: Send Repository Dispatch Event (RC redeploy)
-        if: steps.check-rc-tag.outputs.match == 'true'
+        if: needs.ParseTags.outputs.rc_tag == 'true'
```

Comment:
i think we need to move this out into its own separate job, or else it will be triggered by each published application image within the matrix, and that's too many dispatch notifications; it should just be one notification when all application images have been pushed. i'll push up a commit to adjust this. i think i'd gotten this wrong in my original PR, sorry for that!

Reply:
updated in [5b46063](https://github.com/ethyca/fides/pull/4024/commits/5b46063e2af95ea168cd5765619c75efacb53c84)